app/components/ui/SiteSearchInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FormEvent } from "react";
+import { useHistory } from "react-router-dom";
+import cn from "classnames";
+import qs from "qs";
+import { useSiteSearch } from "../../hooks/SiteSearch";
+import styles from "./SiteSearchInput.module.scss";
+
+/**
+ * Sitewide listing search component
+ *
+ * - Updates the url querystring on every search
+ * - Allows empty searches by removing `query=` query param from querystring
+ */
+export const SiteSearchInput = () => {
+  const { query, setQuery } = useSiteSearch();
+  const history = useHistory();
+
+  const removeQueryStringStart = (querystring: string) => querystring.slice(1);
+
+  const submitSearch = (e: FormEvent) => {
+    e.preventDefault();
+    const queryString = removeQueryStringStart(window.location.search);
+    const searchState = qs.parse(queryString);
+
+    if (query) {
+      searchState.query = query;
+    } else {
+      delete searchState.query;
+    }
+
+    history.push(`/search?${qs.stringify(searchState)}`);
+    return false;
+  };
+
+  return (
+    <form
+      onSubmit={submitSearch}
+      className={cn([styles.navSearch, "search-container", "form-row"])}
+      role="search"
+    >
+      <button
+        type="button"
+        aria-label="Search input icon"
```

Comment:
:accessibility:

Reply:
type should be "submit" and aria-label also "Submit" (aria-label is what the screen reader reads aloud so generally just a straight forward description of what the button does not necessarily what's on it)